src/lib.rs

```diff
@@ -135,12 +134,13 @@ macro_rules! algorithm {
         }
         #[no_mangle]
         #[link_section = ".entry"]
-        pub unsafe extern "C" fn ProgramPage(addr: u32, size: u32, data: *const u8) -> u32 {
+        pub unsafe extern "C" fn ProgramPage(addr: u32, size: u32, buf: *const u8) -> u32 {
```

Comment:
```suggestion
        pub unsafe extern "C" fn ProgramPage(addr: u32, size: u32, data: *const u8) -> u32 {
```

I rather keep this called `data` :)